### PR TITLE
Adapt sdk-test dependency overrides for non-nullable analyzer

### PR DESCRIPTION
**Pull request by jcollins-g**
Deals with #2500 until the new analyzer is published, at least.

### CODE REVIEW

**Comment by parlough** [comment]
Could we instead just expand the normal dependency constraints?

**Comment by jcollins-g** [comment]
> Could we instead just expand the normal dependency constraints?

No.  I actually started down that road earlier -- it turns out that several of the intermediate packages dartdoc depends on have constraints of their own.  For now, it seems best to submit this, then work on fixing those other packages so we can expand our constraints successfully.

**Review by parlough — APPROVED** [review]
Looks good to me

File: tool/grind.dart [hunk 1]
```diff
@@ -598,6 +605,7 @@ Future<void> startTestPackageDocsServer() async {
   log('launching dhttpd on port 8002 for SDK');
   var launcher = SubprocessLauncher('serve-test-package-docs');
   await launcher.runStreamed(sdkBin('pub'), [
+    'global',
```
Comment (srawlins) [inline tool/grind.dart:608]: clever fix

**Review by srawlins — APPROVED** [review]

**Pull request merged**